spec/services/bulk_upload/lettings/year2024/row_parser_spec.rb

```diff
@@ -104,8 +104,6 @@
       "POSTCODE": "SE2 6RT"
 
          }}]}', headers: {})
-
-      parser.valid?
```

Comment:
This was causing problems in some tests - the validation had already been run before anything put in a context-specific before hook, which meant some tests were not actually in the setup they looked like they were in.

Reply:
I've essentially just moved the `parser.valid?` call inside all the relevant tests. I've made a few tweaks to a couple of tests that had issues, but left most things alone due to sheer volume.